### ISSUE TITLE
Automate fetching of new plans and updating model online

### ISSUE THREAD

**Issue by joegoldbeck** [techforwarren/warren-plan-bot#137]
We're getting to the point where plans are coming out and users are trying to use them before we can add the plans.

We can integrate with the [search api](https://ew-search-api.herokuapp.com/search?query=""&index=content&collections=plans-en-US) and use that to discover new plans _and_ pull down their summaries automatically (if we're switching them based on #136 ). We'd then need to update the models as well, but we can do that as well.

This would require a pretty major shift, however. We'd either need to

- have the job commit to the GitHub repo
- store all the plans and summaries in a database

The downside to using the database is that our labeled plans and such are all local, and having that local reference is nice. We'd need to move quite a bit more out of the repo and into data stores to have it be consistent.

**Comment by Hucxley** [comment]
What if we only store the plan URLs (or another unique identifier) of the plans we've already added to our model, and only pull, process and update new plans until we get a chance to add them to plans.json locally?

Once they're added to the plans.json file, it would process locally, but until then it would use the remote plan. Is that a workable alternative that is less disruptive?

**Comment by Hucxley** [comment]
I'm silly.  We don't need to store the plans we already have added to a db, we _basically_ already have one in plans.json.  

We just need to check the URL property of the entries in plans.json against the URLs of the plan data results from the search API, and if there's a plan with a URL that we don't have in plans.json, we use the "temporary" process with our matching algo until we can get the new plan added to our local plans file.
